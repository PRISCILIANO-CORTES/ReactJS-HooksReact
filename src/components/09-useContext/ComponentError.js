import React from 'react'

export const ComponentError = () => {
    return (
      <div>
          <h5>Has llegado a un mundo desconocido</h5>
      </div>
    )
}
