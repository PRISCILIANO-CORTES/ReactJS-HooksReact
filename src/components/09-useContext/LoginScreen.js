import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    // 1. Obtener la referencia del context user
    const { user } = useContext(UserContext);

    console.log(user);

    // 2. Obtener la referencia  del context setUser
    const { setUser } = useContext(UserContext);

    console.log(setUser);

    const newUser = {
        id: 12345,
        name: 'Prisciliano'
    }


    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className='btn btn-primary btn-block'
                onClick={ () => setUser(newUser)}
            >
                Login
            </button>
        </div>
    )
}
