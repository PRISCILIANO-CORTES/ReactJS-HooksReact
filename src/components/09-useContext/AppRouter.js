// RUTAS CON REACT-ROUTER-DOM 

import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route,
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { ComponentError } from './ComponentError';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className='container'>
                    <Routes>
                        <Route path="/" exact element={ <HomeScreen /> } />
                        <Route path="/about" exact element={ <AboutScreen /> } />
                        <Route path="/login" exact element={ <LoginScreen /> } />
                        <Route path="*" element={ <ComponentError /> } />
                    </Routes>
                </div>
            </div>
        </Router>
    )
}
