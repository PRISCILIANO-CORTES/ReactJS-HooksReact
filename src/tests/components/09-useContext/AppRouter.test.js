import { mount } from "enzyme";
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('Pruebas en <AppRouter />', () => {

    const user = {
        name: 'Prisciliano',
        email: 'pris98.cortesgmail.com'
    }

    const wrapper = mount( 
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter /> 
        </UserContext.Provider>
    );

    test('Debe mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    })
})