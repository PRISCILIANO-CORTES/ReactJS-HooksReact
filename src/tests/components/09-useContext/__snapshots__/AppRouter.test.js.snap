// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      UseContext
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link active"
          exact={true}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          activeClassName="active"
          className="nav-item nav-link"
          exact={true}
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          activeClassName="active"
          className="nav-item nav-link"
          exact={true}
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Prisciliano",
   "email": "pris98.cortesgmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
